scRNA-seq/05-dimension_reduction_scRNA-seq.Rmd

```diff
@@ -6,234 +6,411 @@ output:
-tsne_data <- data.frame(tsne_tab_mur$Y)
-  
-# Plot this with ggplot2
-ggplot(tsne_data, aes(x = X1, y = X2)) + 
-  geom_point() + 
-  colorblindr::scale_color_OkabeIto()
+ggplot(pca_tab_mur, 
+       aes(x = PC1, y = PC2)) + 
+  geom_point() 
 ```
 
-#### Apply metadata labels to our t-SNE
+It looks like there are a few different clusters here, which may represent different kinds of cells within the bladder tissue.
+If we wanted to, we might look at marker genes within these clusters to try to determine which kinds of cells are in each cluster from this tissue. 
+This is exactly what the *Tabula Muris* group ended up doing, and you can see some of their results along that front at https://tabula-muris.ds.czbiohub.org/.
 
-Let's re-plot this with labels using some of our metadata to see 
-how the cells are clustering (or not clustering) by different variables. 
-Remember that `tab_mur_meta` has the corresponding information for each of our
-cells. 
-Below we have set up the cells to be colored by the `tissue` variable. 
-Run this, but also feel free to look at what other information we have for these
-cells in `tab_mur_meta` by using `str()` and change this to a different variable to 
-graph it by.
 
-```{r}
-# Plot this with ggplot2, but add the color argument
-ggplot(tsne_data, aes(x = X1, y = X2, color = tab_mur_meta$tissue)) + 
-  geom_point() +
-  colorblindr::scale_color_OkabeIto()
+## Beyond PCA: UMAP
+
+### More from the *Tabula Muris* Dataset
+
+To explore more about dimensionality reduction, we will explore a larger subset of the data from the *Tabula Muris* study. 
+We have run alevin on 15 samples, representing 9 different tissues, and taken each of those samples through QC and normalization similar to what we have presented above (with slightly different cutoffs).
+Following that, we selected the 1000 genes with the greatest biological variance (using the same method as shown in `02-normalizing_scRNA-seq.Rmd`).
+
+These results, with accompanying metadata were saved into a `SingleCellExperiment` object, which is stored as RDS file at the location `tm_normalized_file` (defined in the `filepaths` chunk above).
+Let's load that in and have a quick look at it. 
+
+```{r load_normalized}
+tm_normalized <- readr::read_rds(tm_normalized_file)
 ```
 
-Now that we have an idea of what the default parameters look like, let's try
-experimenting with the perplexity parameter. 
-For us to determine a good number range to use for testing perplexity, let's 
-first look up the documentation for `Rtsne::Rtsne` and find out what the 
-default for the `perplexity` argument is. 
+How many cells are in this data? How many samples are there? How many cells from each tissue?
 
-In order to prepare for this experiment, we will create a *function* that allows
-us to rerun this same code chunk easily, but create an argument that allows us
-to change one variable: our perplexity variable. 
+```{r explore_sce, live = TRUE}
```

Comment:
I also think delivery would be easier if these were different chunks. This may be personal preference. (The completed version would also be a little easier to follow in my opinion.)

Reply:
This is another place where my plan was to set them free a bit. The code below is example code... I agree that separate chunks would be preferable for the completed version, but didn't want a bunch of blank chunks.